feat: Add docs to install Nginx Gateway Fabric with N+ (#1524)
nginxinc/nginx-gateway-fabric: site/content/installation/installing-ngf/manifests.md

Deploying NGINX Gateway Fabric with Kubernetes manifests takes only a few steps. With manifests, you can configure your deployment exactly how you want. Manifests also make it easy to replicate deployments across environments or clusters, ensuring consistency.

- If you’d like to use NGINX Plus:

>   - To pull from the F5 Container registry, configure a docker registry secret using your JWT token from the MyF5 portal by following the instructions from [here](https://docs.nginx.com/nginx-gateway-fabric/installation/ngf-images/jwt-token-docker-secret). Make sure to specify the secret using `nginxGateway.serviceAccount.imagePullSecretName` or `nginxGateway.serviceAccount.imagePullSecretsNames` parameter.

Reviewer: ```suggestion
  - To pull from the F5 Container registry, configure a docker registry secret using your JWT token from the MyF5 portal by following the instructions from [here](https://docs.nginx.com/nginx-gateway-fabric/installation/ngf-images/jwt-token-docker-secret). Make sure to specify the secret in the `imagePullSecrets` field of the `nginx-gateway` ServiceAccount.
```

>   1. To pull from the F5 Container registry, configure a docker registry secret using your JWT token from the MyF5 portal by following the instructions from [here](https://docs.nginx.com/nginx-gateway-fabric/installation/ngf-images/jwt-token-docker-secret). Make sure to specify the secret using `nginxGateway.serviceAccount.imagePullSecret` or `nginxGateway.serviceAccount.imagePullSecrets` parameter.
>   1. Update the `image` field to `nginx-gateway-fabric/nginx-plus:edge` in the manifest.

Reviewer: ```suggestion
  1. To pull from the F5 Container registry, configure a docker registry secret using your JWT token from the MyF5 portal by following the instructions from [here](https://docs.nginx.com/nginx-gateway-fabric/installation/ngf-images/jwt-token-docker-secret). Make sure to specify the secret in the `imagePullSecrets` field of the `nginx-gateway` ServiceAccount.
  1. Alternatively, pull an NGINX Gateway Fabric image with NGINX Plus and push it to your private registry by following the instructions from [here]({{<relref "installation/ngf-images/pulling-ngf-image.md">}}).
  1. Update the nginx container's `image` field of the `nginx-gateway` Deployment accordingly.
```

>   - To generate manifests for NGINX Plus, use `make generate-manifests-plus` at the root directory of the repository. You should see `image` field updated with `nginx-gateway-fabric/nginx-plus:edge`.

Reviewer: You can still mention the alternative option of pulling the image and pushing to a private registry, like you do in the helm guide.

Regarding generating the manifests, since we are committing them, a user shouldn't need to do that. Instead, you can just mention that they should update the image field in the manifest.

Reviewer: You should mention in the installation steps below how to install using the plus manifests instead of the regular manifests, similar to how the helm guide has two sets of instructions.

Deploying NGINX Gateway Fabric with Kubernetes manifests takes only a few steps. With manifests, you can configure your deployment exactly how you want. Manifests also make it easy to replicate deployments across environments or clusters, ensuring consistency.

- If you’d like to use NGINX Plus:

>   - To pull from the F5 Container registry, configure a docker registry secret using your JWT token from the MyF5 portal by following the instructions from [here](https://docs.nginx.com/nginx-gateway-fabric/installation/ngf-images/jwt-token-docker-secret). Make sure to specify the secret using `nginxGateway.serviceAccount.imagePullSecretName` or `nginxGateway.serviceAccount.imagePullSecretsNames` parameter.

Author: removing & updating

>   - To pull from the F5 Container registry, configure a docker registry secret using your JWT token from the MyF5 portal by following the instructions from [here](https://docs.nginx.com/nginx-gateway-fabric/installation/ngf-images/jwt-token-docker-secret). Make sure to specify the secret using `nginxGateway.serviceAccount.imagePullSecretName` or `nginxGateway.serviceAccount.imagePullSecretsNames` parameter.

Author: updated

- If you’d like to use NGINX Plus:
  - To pull from the F5 Container registry, configure a docker registry secret using your JWT token from the MyF5 portal by following the instructions from [here](https://docs.nginx.com/nginx-gateway-fabric/installation/ngf-images/jwt-token-docker-secret). Make sure to specify the secret using `nginxGateway.serviceAccount.imagePullSecretName` or `nginxGateway.serviceAccount.imagePullSecretsNames` parameter.
  - Alternatively, pull an Nginx Gateway Fabric image with NGINX Plus and push it to your private registry by following the instructions from [here]({{ ref "installation/ngf-images/pulling-ngf-image.md" }}).
  - Alternatively, you can build an Nginx Gateway Fabric image with NGINX Plus and push it to your private registry by following the instructions from [here]({{< relref "installation/building-the-images.md" >}}).
  - Update the `nginxGateway.image.repository` field of the `values-plus.yaml` accordingly.

> 

Reviewer: These configuration options don't apply to manifests, only helm. We should add steps to the `make generate-manifests` target to also generate manifests for an NGINX Plus installation. NIC has manifests for both OSS and Plus, so we should too.

Author: as part of this PR?

Reviewer: Yes, it should be in the ticket. Our docs will reference manifests to install, so they should exist either before or while we add those docs.

Author: okay will do so

Author: updated makefile to generate plus manifests

#### Edge version

>    1. For NGINX:

Reviewer: Let's use bullet points instead of numbered list for these two options.

   ```shell
   kubectl apply -f deploy/manifests/nginx-gateway.yaml
   ```

   1. For NGINX Plus

   ```shell
   kubectl apply -f deploy/manifests/nginx-plus-gateway.yaml
   ```

>    Update the nginx-plus-ingress.yaml file to include your chosen image from the F5 Container registry or your custom container image.

Reviewer: ```suggestion
   Update the nginx-plus-gateway.yaml file to include your chosen image from the F5 Container registry or your custom container image.
```
